scheduler/scheduler-server/src/test/java/functionaltests/job/TestJobWhenSchedulerPaused.java

```diff
@@ -0,0 +1,153 @@
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package functionaltests.job;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.net.URL;
+
+import org.junit.Test;
+import org.ow2.proactive.scheduler.common.Scheduler;
+import org.ow2.proactive.scheduler.common.SchedulerState;
+import org.ow2.proactive.scheduler.common.job.JobId;
+import org.ow2.proactive.scheduler.common.job.JobState;
+import org.ow2.proactive.scheduler.common.job.JobStatus;
+
+import functionaltests.utils.SchedulerFunctionalTestNoRestart;
+
+
+/**
+ * Test jobs status when the scheduler is paused
+ */
+public class TestJobWhenSchedulerPaused extends SchedulerFunctionalTestNoRestart {
+
+    private static String simpleJobPath = "/functionaltests/descriptors/Job_simple.xml";
+
+    private static String sleep5sJobPath = "/functionaltests/descriptors/Job_5s.xml";
+
+    private static URL simpleJobUrl = TestJobWhenSchedulerPaused.class.getResource(simpleJobPath);
+
+    private static URL sleep5sJobUrl = TestJobWhenSchedulerPaused.class.getResource(sleep5sJobPath);
+
+    private final static int EVENT_TIMEOUT = 5000;
+
+    @Test
+    public void testJobsDontRunWhenSchedulerPaused() throws Throwable {
+
+        // Get the initial job number
+        int initialJobsNumber = refreshAndGetTotalJobCount();
+
+        // Pause the scheduler
+        Scheduler schedAdminInterface = schedulerHelper.getSchedulerInterface();
+        assertTrue(schedAdminInterface.pause());
+
+        // Ensure that every job newly submitted will be pending
+        JobId jobId1 = schedulerHelper.submitJob(new File(simpleJobUrl.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(simpleJobUrl.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventJobSubmitted(jobId1, EVENT_TIMEOUT);
+        schedulerHelper.waitForEventJobSubmitted(jobId2, EVENT_TIMEOUT);
+
+        // Without sleeping, we could could test if jobs are blocked in pending state ..
+        // .. without waiting long enough that they run
+        Thread.sleep(10000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)